src/backend/feed.js

```diff
@@ -0,0 +1,33 @@
+const storage = require('./utils/storage');
+
+/**
+ * Feed Object, upon being created all feeds will be active and have a feedId of 0 until saved into Redis
+ * @param url url of feed
+ * @param name author name of feed
+ * @param feedId unique id for feed
+ */
+class Feed {
+  constructor(url, name) {
+    this.url = url;
+    this.name = name;
+    this.inactive = false;
```

Comment:
I don't think we should include this in the actual data.  Let's have it be something that gets exposed based on which set it's in.  When we read from the db, we'll figure that out.

Reply:
so in this case we just want the this.inactive as the only property?

Reply:
It's going to have to be an async getter/setter, since it requires a database operation vs. storing something on `this`.  We don't need an actual property, just a way to get at the database.

Reply:
Got it, don't we have a problem here then? If we're trying to retrieve from the database, should active/inactive be kept track of in the DB as well?

Reply:
My thought here is that the state we're going to store won't be on the Object, but rather, in which of the two sets this hash object lives in Redis.  A feed is either in the feeds or inactive-feeds set, but not both.  So we can use that as a state we read from the database, and expose as a boolean via the JS API.

What about:

```js
const isActive = await feed.isActive(); // checks if this feed is in feeds set
await feed.inactive(); // set this feed inactive (move to inactive feeds)
await feed.active(); // set this feed active (move to feeds)
```

Reply:
I guess what I'm confused about is I have code like this now

```
async inactive() {
  const isActive = await this.isActive();
  await this.save();
}
```
I won't be passing isActive to save(). So now my addFeed(feed) function which is called when save() is called can't differentiate between whether it should add it to FEEDS or INACTIVE_FEEDS. The only workaround I can think about this is to separate the addFeed function into a addActiveFeed(feed) and addInactiveFeed(feed).